### DISCUSSION TITLE
isLoading ignoring `fallback`

### DISCUSSION THREAD

**Discussion by Dragate** [vercel/swr#2291] [Q&A]
# Bug report

## Description / Observed Behavior

`isLoading` becomes true on the first request even though data is already available for the given key through SWRConfig's `fallback`, which makes the UI show a loading indicator for the first revalidation (as if `isValidating` was used).

## Expected Behavior

Expected `isLoading` to remain false when initial data for a given url/key is provided from server (getServerSideProps/getStaticProps).

## Repro Steps / Code Example

[CodeSandbox](https://codesandbox.io/s/amazing-glade-mch76x)
Trigger page reload in CSB to see the loading indicator.

## Additional Context

swr@2.0.0

Documentation says 

> Fallback data and previous data are not considered "loaded data"

although I'm not sure if this includes `fallback` or only `fallbackData`. 

Previously was using https://swr.vercel.app/docs/middleware#keep-previous-result but wanted to reduce code with the new `isLoading` and `keepPreviousData`. Should I keep using the middleware?

**Comment by koba04** [comment]
As the documentation says, this is an expected behavior.
https://swr.vercel.app/docs/advanced/understanding#fallback

I'd say say you can check `data === undefined` for that case.

**Reply by Dragate** [reply]
`data === undefined` wouldn't work, as in my case data is always present. It's pre-populated from getServerSideProps (using `fallback`) and `keepPreviousData` makes sure to persist data until the next key provides another set of data.

In any case, thanks for confirming the intended behaviour. I'll stick to the middleware then.
